File: 1° Período - Introdução a Programação 1/Exercicios em Python/Lista 5 - Python/q4.py
#Faça um algoritmo que leia um número inteiro e calcule o seu fatorial. Se o número for negativo,
#informe que o valor é inválido.

x=int(input("N: "))
i=1
n_fat=1

if x<0:
    print("Número inválido!")
else:
    while i<=x:
        n_fat=n_fat*i
        i=i+1
    print("Fatorial: ", n_fat)
    

File: 1° Período - Introdução a Programação 1/Exercicios em C/Lista 5 - C/q4.c
//. Faça um algoritmo que leia um número inteiro e calcule o seu fatorial. Se o número for negativo,
//informe que o valor é inválido.
#include<stdio.h>
int main(){
	int x, n_fat, i;
	i=1;
	n_fat=1;
	printf("Insira um numero: ");
	scanf("%d", &x);
	if(x<0){
		printf("Número inválido!");
	}
	else{
		while(i<=x){
			n_fat=n_fat*i;
	        i=i+1;
		}
		printf("Fatorial: %d", n_fat);	     
	}  
}
